tests/integration/test_backup_restore_new/test.py

```diff
@@ -1798,26 +1798,25 @@ def test_system_backups():
     id = instance.query(
         f"RESTORE TABLE test.table FROM {backup_name}", query_id=restore_query_id
     ).split("\t")[0]
-    restore_info = get_backup_info_from_system_backups(by_id=id)
-    restore_events = get_events_for_query(restore_query_id)
 
-    assert restore_info.name == escaped_backup_name
-    assert restore_info.status == "RESTORED"
-    assert restore_info.error == ""
-    assert restore_info.start_time < restore_info.end_time
-    assert restore_info.num_files == info.num_files
-    assert restore_info.total_size == info.total_size
-    assert restore_info.num_entries == info.num_entries
-    assert restore_info.uncompressed_size == info.uncompressed_size
-    assert restore_info.compressed_size == info.compressed_size
-    assert (
-        restore_info.files_read + restore_events["RestorePartsSkippedFiles"]
-        == restore_info.num_files
-    )
-    assert (
-        restore_info.bytes_read + restore_events["RestorePartsSkippedBytes"]
-        == restore_info.total_size
-    )
+    def verify_restore_info():
+        restore_info = get_backup_info_from_system_backups(by_id=id)
+        restore_events = get_events_for_query(restore_query_id)
+        return (
+            restore_info.name == escaped_backup_name and
+            restore_info.status == "RESTORED" and
+            restore_info.error == "" and
+            restore_info.start_time < restore_info.end_time and
+            restore_info.num_files == info.num_files and
+            restore_info.total_size == info.total_size and
+            restore_info.num_entries == info.num_entries and
+            restore_info.uncompressed_size == info.uncompressed_size and
+            restore_info.compressed_size == info.compressed_size and
+            restore_info.files_read + restore_events["RestorePartsSkippedFiles"] == restore_info.num_files and
+            restore_info.bytes_read + restore_events["RestorePartsSkippedBytes"] == restore_info.total_size
+        )
+
+    wait_condition(verify_restore_info, lambda x: x);
```

Comment:
It looks like the default attempts in 10 with 0.1 delay - probably okay to start with and adjust later on.
https://github.com/ClickHouse/ClickHouse/blob/80e8f168c83280c1284d90dcb69ed42164d9b1e4/tests/integration/helpers/test_tools.py#L187

Reply:
Yeah, total of 1s seems like a pretty decent amount of time for updates to flow through. In the test which fails, the timestamps are as follows (so, within 0.1 seconds of each other):

```
2025.02.09 00:32:27.595741 [ 696 ] {c403802e95af4d318426d4228e3fa598} <Information> BackupsWorker: Restored from backup Disk('backups', '50/') successfully
...
2025.02.09 00:32:27.644105 [ 10 ] {b0682c70-373c-4d33-83eb-94889d13d93d} <Debug> executeQuery: (from 172.16.3.1:60550) (query 1, line 1) SELECT name, id, status, error, num_files, total_size, num_entries, uncompressed_size, compressed_size, files_read, bytes_read FROM system.backups WHERE id = '796cf0de-1613-4f43-a29d-01e0634656b9' LIMIT 1 (stage: Complete)
```